packages/cli/src/daemon/DaemonDecider.ts

```diff
@@ -142,28 +146,29 @@ export class DaemonDecider {
      * @memberof DaemonDecider
      */
     private initialParse() {
-        const numOfParms = this.mParms.length - 2;
-        this.mPort = DaemonDecider.DEFAULT_PORT;
-
-        if (numOfParms > 0) {
-            const parm = this.mParms[2];
-
+        if (this.mParms.length > 2) {
             /**
-             * NOTE(Kelosky): For now, we use an undocumented paramter `--daemon`.  If found first,
+             * NOTE(Kelosky): For now, we use an undocumented parameter `--daemon`.  If found first,
              * we bypass `yargs` and begin running this as a persistent Processor.
              */
-            const portOffset = parm.indexOf(DaemonDecider.DAEMON_KEY);
+            const parm = this.mParms[2];
+            const daemonOffset = parm.indexOf(DaemonDecider.DAEMON_KEY);
 
-            if (portOffset > -1) {
+            if (daemonOffset > -1) {
                 this.startServer = true;
+
                 if (process.env.ZOWE_DAEMON) {
-                    try {
-                        this.mPort = parseInt(process.env.ZOWE_DAEMON, 10);
-                    } catch (err) {
-                        // do nothing
+                    this.mSocket = process.env.ZOWE_DAEMON;
+                    if (process.platform === "win32") {
+                        this.mSocket = `\\\\.\\pipe\\${this.mSocket}`;
                     }
+                } else if (process.platform !== "win32") {
+                    this.mSocket = path.join(os.homedir(), ".zowe-daemon.sock");
+                } else {
+                    this.mSocket = `\\\\.\\pipe\\${os.userInfo().username}\\ZoweDaemon`;
                 }
-                Imperative.api.appLogger.debug(`daemon server port ${this.mPort}`);
+
+                Imperative.api.appLogger.debug(`daemon server socket ${this.mSocket}`);
```

Comment:
Should we be a bit more generic here ?
instead of socket, maybe say Communication Mechanism (since we are using pipes for win32) ?